### PR TITLE
Fix a possible nil-dereference crash

### PR DESCRIPTION
**Pull request by krnowak**
If we get a bounded instrument from an instrument with an already set
delegate and then concurrently call Unbind and RecordOne on it, it is
possible that Unbind will trigger the sync.Once "initialize" member
with a no op function, so the sync.Once "initialize" in RecordOne will
not be triggered anymore. This in turn will not set the implPtr to the
real bound instrument, so an atomic load later in RecordOne will again
return a nil pointer which we unconditionally dereference.

Add an extra check for a nil pointer - if this is true, then Unbind
was first and RecordOne should effectively be a no op.

### CODE REVIEW

File: api/global/internal/meter.go [hunk 1]
```diff
@@ -233,6 +233,9 @@ func (bound *instHandle) RecordOne(ctx context.Context, number core.Number) {
 	if implPtr == nil {
 		implPtr = (*metric.BoundInstrumentImpl)(atomic.LoadPointer(&bound.delegate))
 	}
+	if implPtr == nil {
+		return
```
Comment (lizthegrey) [inline api/global/internal/meter.go]: this needs comments explaining why/how this could occur, similar to what you've put in the PR description.

Comment (krnowak) [reply api/global/internal/meter.go]: Done.

**Comment by krnowak** [comment]
Updated, added the test.

File: api/global/internal/meter_test.go [hunk 2]
```diff
@@ -225,3 +225,23 @@ func TestDefaultSDK(t *testing.T) {
 	require.Equal(t, `{"updates":[{"name":"test.builtin{A=B}","sum":1}]}
 `, <-ch)
 }
+
+func TestUnbindThenRecordOne(t *testing.T) {
+	internal.ResetForTest()
+
+	// Note: this test uses oppsite Float64/Int64 number kinds
```
Comment (jmacd) [inline api/global/internal/meter_test.go]: Delete this comment.

Comment (krnowak) [reply api/global/internal/meter_test.go]: Oops, copy pasta. Will fix in a separate PR.

**Pull request merged**